azkaban-common/src/main/java/azkaban/executor/ExecutionController.java

```diff
@@ -89,8 +90,13 @@ public long getLastExecutorManagerThreadCheckTime() {
 
   @Override
   public Collection<Executor> getAllActiveExecutors() {
-    // Todo: get the executor info from DB
-    return Collections.emptyList();
+    List<Executor> executors = new ArrayList<>();
+    try {
```

Comment:
I'm trying to understand why the exception should be cached in this block. Any particular reason?

Reply:
Added an additional check for empty executor list in the new commit.